components/Seo.php

```diff
@@ -58,4 +36,152 @@ public function defineProperties()
+        } else {
+            $properties = array_merge(
+                $this->page['viewBag']->getProperties(), $this->page->settings
+            );
+
+            $this->viewBagProperties = $properties;
+            $this->page['viewBag']->setProperties($properties);
+        }
+        $this->disable_schema = $this->property('disable_schema');
+        $this->locale = App::getLocale();
+    }
+
+    public function getTitle()
+    {
+        $title = $this->viewBagProperties['title'];
+        if (isset($this->viewBagProperties['meta_title'])) {
+            $title = $this->viewBagProperties['meta_title'];
+        }
+
+        if (isset($this->viewBagProperties['meta_title[' . $this->locale . ']'])) {
+            $title = $this->viewBagProperties['meta_title[' . $this->locale . ']'];
+        }
+
+        $settings = Settings::instance();
+
+        if ($settings->site_name_position == 'prefix') {
+            $title = "{$settings->site_name} {$settings->site_name_separator} {$title}";
+        } else if ($settings->site_name_position == 'suffix') {
+            $title = "{$title} {$settings->site_name_separator} {$settings->site_name}";
+        }
+
+        return $title;
+    }
+
+    public function getDescription()
+    {
+        $description = Settings::instance()->description;
+        if (isset($this->page['description'])) {
+            $description = $this->page['description'];
+        }
+
+        if (isset($this->viewBagProperties['meta_description'])) {
+            $description = $this->viewBagProperties['meta_description'];
+        }
+
+        if (isset($this->viewBagProperties['localeMeta_description[' . $this->locale . ']'])) {
+            $description = $this->viewBagProperties['localeMeta_description[' . $this->locale . ']'];
+        }
+        return $description;
+    }
+
+    public function getOgTitle()
+    {
+        $ogTitle = $this->getTitle();
+        if (isset($this->viewBagProperties['og_title'])) {
+            $ogTitle = $this->viewBagProperties['og_title'];
+        }
+
+        if (isset($this->viewBagProperties['og_title[' . $this->locale . ']'])) {
+            $ogTitle = $this->viewBagProperties['og_title[' . $this->locale . ']'];
+        }
+        return $ogTitle;
+    }
+
+    public function getOgDescription()
+    {
+        $ogDescription = $this->getDescription();
+        if (isset($this->viewBagProperties['og_description'])) {
+            $ogDescription = $this->viewBagProperties['og_description'];
+        }
+
+        if (isset($this->viewBagProperties['LocaleOg_description[' . $this->locale . ']'])) {
+            $ogDescription = $this->viewBagProperties['LocaleOg_description['. $this->locale . ']'];
+        }
+        return $ogDescription;
+    }
+
+    public function getOgImage()
+    {
+        $mediaUrl = url(Config::get('cms.storage.media.path'));
+        $ogImage = $this->getSiteImageFromSettings();
+        if ($settingsSiteImage = Settings::instance()->siteImage) {
```

Comment:
This place can also make use of the `??` operator which will make it 3 times easier and shorter.